### PR TITLE
Feature delete

### PR DESCRIPTION
**Pull request by BigheartSmallbrain**
Fix delete command such that undo command will restore all events deleted instead of just previously mentioned one event.

Also undo command will be able to undo previous command when the delete command is invalid.

### CODE REVIEW

File: src/main/java/ezschedule/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -35,27 +35,27 @@ public DeleteCommand(List<Index> targetIndexes) {
...
         }
+        model.clearRecent();
+        for (Index targetIndex: targetIndexes) {
+            model.addRecentCommand(this);
+            Event eventToDelete = lastShownList.get(targetIndex.getZeroBased());
+            model.deleteEvent(eventToDelete);
+            model.addRecentEvent(eventToDelete);
+            feedback.insert(0, eventToDelete.toString());
+        }
         feedback.insert(0, MESSAGE_DELETE_EVENT_SUCCESS);
```
Comment (JrmCkh) [inline src/main/java/ezschedule/logic/commands/DeleteCommand.java:59]: why is there a need for double for loop?

File: src/main/java/ezschedule/model/Model.java [hunk 2]
```diff
@@ -97,6 +97,10 @@ public interface Model {
      * Adds the given event as recent event.
      */
     void addRecentEvent(Event event);
+    /**
+     * Adds the given command as most recently executed command
+     */
+    void addRecentCommand(Command command);
```
Comment (JrmCkh) [inline src/main/java/ezschedule/model/Model.java:104]: pls add spacing in line 100 (for readability and code quality)

File: src/main/java/ezschedule/model/ModelManager.java [hunk 3]
```diff
@@ -158,6 +158,10 @@ public ArrayList<Event> recentEvent() {
     public void addRecentEvent(Event event) {
         recentEvent.add(event);
     }
+    @Override
+    public void addRecentCommand(Command command) {
+        recentCommand.add(command);
+    }
```
Comment (JrmCkh) [inline src/main/java/ezschedule/model/ModelManager.java:165]: pls add spacing in line 161 (for readability and code quality)

File: src/test/java/ezschedule/logic/commands/AddCommandTest.java [hunk 4]
```diff
@@ -176,6 +176,10 @@ public ArrayList<Event> recentEvent() {
         public void addRecentEvent(Event event) {
             throw new AssertionError("This method should not be called.");
         }
+        @Override
+        public void addRecentCommand(Command command) {
+            throw new AssertionError("This method should not be called");
+        }
```
Comment (JrmCkh) [inline src/test/java/ezschedule/logic/commands/AddCommandTest.java:183]: pls add spacing in line 179 (for readability and code quality)

File: src/main/java/ezschedule/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (BigheartSmallbrain) [reply src/main/java/ezschedule/logic/commands/DeleteCommand.java:59]: First loop is to check and ensure that the index provided is valid and doesnt throw any exception.
So in the case which first loop are passed, it suggests that all index provided are valid and proceed on to clearRecent() once and adding all the events to delete into the recentEvent list.

**Review by JrmCkh — APPROVED** [review]
LGTM

**Pull request merged**